lib/utils/video_player/src/thunder_youtube_player.dart

```diff
@@ -89,6 +93,40 @@ class _ThunderYoutubePlayerState extends State<ThunderYoutubePlayer> with Single
     return false;
   }
 
+  int? extractYouTubeTimestamp() {
+    final uri = Uri.parse(widget.videoUrl);
+
+    // Check for timestamp in query parameters
+    String? timeParam = uri.queryParameters['t'] ?? uri.queryParameters['start'];
+
+    // Check for embedded timestamps like t=1m30s
+    final regex = RegExp(r't=(\d+h)?(\d+m)?(\d+s)?');
+
+    if (timeParam != null) {
+      return _convertTimeStringToSeconds(timeParam);
+    } else if (regex.hasMatch(widget.videoUrl)) {
+      final match = regex.firstMatch(widget.videoUrl)!;
+      return _convertTimeComponentsToSeconds(match);
+    }
+
+    return null;
+  }
+
+  int _convertTimeStringToSeconds(String time) {
+    final regex = RegExp(r'(\d+h)?(\d+m)?(\d+s)?');
+    final match = regex.firstMatch(time);
+
+    return match != null ? _convertTimeComponentsToSeconds(match) : 0;
+  }
+
+  int _convertTimeComponentsToSeconds(RegExpMatch match) {
+    int hours = match.group(1) != null ? int.parse(match.group(1)!.replaceAll('h', '')) : 0;
+    int minutes = match.group(2) != null ? int.parse(match.group(2)!.replaceAll('m', '')) : 0;
+    int seconds = match.group(3) != null ? int.parse(match.group(3)!.replaceAll('s', '')) : 0;
+
+    return hours * 3600 + minutes * 60 + seconds;
```

Comment:
You could do this as `Duration(hours: hours, minutes: minutes, seconds: seconds).inSeconds` to avoid "magic numbers".